### PR TITLE
docs: describe windows dependencies for source builds

### PR DESCRIPTION
**Pull request by ControlCplusControlV**
Sorry I know everyone hates supporting Windows Native but due to some unfortunate circumstances I found myself syncing a reth node in Windows 10, and so thought it might be nice to mention it within the Reth book, and how to install the deps. I did take the dep list from lighthouse, but given reth may be paired with other CL clients and users may not have necessarily read the lighthouse book it should be documented here too

I hope to make more actual PRs to reth in the future when I have more time 😆

### CODE REVIEW

File: book/installation/source.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ operating system:
 
 - **Ubuntu**: `apt-get install libclang-dev pkg-config build-essential`
 - **macOS**: `brew install llvm pkg-config`
+- **Windows**: `choco install make; choco install cmake --installargs 'ADD_CMAKE_TO_PATH=System'; choco install llvm; choco install protoc`
```
Comment (onbjerg) [inline book/installation/source.md:31]: We only need llvm, the rest are not needed

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Comment by gakonst** [comment]
Nothing against Windows installs! Defer to Bjerg here.

File: book/installation/source.md (hunk 1 above)
Comment (ControlCplusControlV) [reply book/installation/source.md]: Fixed

**Review by onbjerg — APPROVED** [review]
Sorry for the delay - small nit (will merge myself), thank you!

**Review by gakonst — COMMENTED** [review]
good one - we love windows sir sir

**Pull request merged**